src/backend/codegen/include/codegen/codegen_manager.h

```diff
@@ -0,0 +1,113 @@
+//---------------------------------------------------------------------------
+//  Greenplum Database
+//  Copyright (C) 2016 Pivotal Software, Inc.
+//
+//  @filename:
+//    codegen_manager.h
+//
+//  @doc:
+//    Object that manage all CodeGen and CodeGenUtils
+//
+//---------------------------------------------------------------------------
+
+#ifndef GPCODEGEN_CODEGEN_MANAGER_H_  // NOLINT(build/header_guard)
+#define GPCODEGEN_CODEGEN_MANAGER_H_
+
+#include <memory>
+#include <vector>
+#include <string>
+
+#include "codegen/utils/macros.h"
+#include "codegen/codegen_wrapper.h"
+
+namespace gpcodegen {
+/** \addtogroup gpcodegen
+ *  @{
+ */
+
+// Forward declaration of CodeGenUtils to manage llvm module
+class CodeGenUtils;
+
+// Forward declaration of a CodeGenInterface that will be managed by manager
+class CodeGenInterface;
+
+/**
+ * @brief Object that manages all code gen.
+ **/
+class CodeGenManager {
+ public:
+  /**
+   * @brief Constructor.
+   *
+   * @param module_name A human-readable name for the module that this
+   *        CodeGenManager will manage.
+   **/
+  explicit CodeGenManager(const std::string& module_name);
+
+  ~CodeGenManager() = default;
+
+  /**
+   * @brief Enroll a code generator with manager
+   *
+   * @note Manager manages the memory of enrolled generator.
+   *
+   * @param funcLifespan Based on life span corresponding CodeGen_Utils will be used to generate
```

Comment:
yes, Let me correct it as 
"Life span of the enrolling generator. Based on life span,
corresponding CodegenUtils will be used for code generation"